core/src/main/java/com/google/cloud/sql/core/CloudSqlInstanceName.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql.core;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CloudSqlInstanceName {
+
+  // Unique identifier for each Cloud SQL instance in the format "PROJECT:REGION:INSTANCE"
+  // Some legacy project ids are domain-scoped (e.g. "example.com:PROJECT:REGION:INSTANCE")
+  private static final Pattern CONNECTION_NAME =
+      Pattern.compile("([^:]+(:[^:]+)?):([^:]+):([^:]+)");
```

Comment:
Let's pull this up and use 100 characters across the board.

Reply:
This line is 102 lines if we make it one line 🙃

Reply:
oops.